source/Engine.cpp

```diff
@@ -790,18 +789,83 @@ void Engine::Step(bool isActive)
 			info.SetCondition("flagship solar display");
 			info.SetString("flagship solar", to_string(flagshipSolar));
 		}
+		// These check for the attribute to determine if the pilot has installed
+		// outfits that give a live display of ship mass and jump fuel costs.
+		bool flagshipMassDisplay = (flagship->DisplayMass());
+		bool flagshipHyperDisplay = (flagship->DisplayHyperFuelCost());
+		bool flagshipScramDisplay = (flagship->DisplayScramFuelCost());
+		bool flagshipJumpDisplay = (flagship->DisplayJumpFuelCost());
+		// These are to get the method of scaling the fuel bar segments.
+		bool flagshipHyperFuelBar = (flagship->HyperDriveFuelBar());
+		bool flagshipScramFuelBar = (flagship->ScramDriveFuelBar());
+		bool flagshipJumpFuelBar = (flagship->JumpDriveFuelBar());
+		double flagshipFixedFuelBar = flagship->FixedScaleFuelBar();
+		// Transfers that information into the info setconditions.
+		if(flagshipMassDisplay)
+			info.SetCondition("flagship mass display");
+		if(flagshipHyperDisplay)
+			info.SetCondition("flagship hyper fuel display");
+		if(flagshipScramDisplay)
+			info.SetCondition("flagship scram fuel display");
+		if(flagshipJumpDisplay)
+			info.SetCondition("flagship jump fuel display");
+		// Calculates the mass, as well as hyper, scram, and jump drive fuel costs
+		double flagshipMass = flagship->InertialMass();
+		int flagshipMassNumber = round(flagshipMass);
+		info.SetString("flagship mass", to_string(flagshipMassNumber));
+		double flagshipHyperDriveFuel = 100 + (((flagshipMass - 900) / 100) * 4);
+		int flagshipHyperDriveFuelDisplay = round(flagshipHyperDriveFuel);
+		info.SetString("flagship hyperdrive fuel per hyperjump", to_string(flagshipHyperDriveFuelDisplay));
+		double flagshipScramDriveFuel = 150 + (((flagshipMass - 900) / 100) * 6);
+		int flagshipScramDriveFuelDisplay = round(flagshipScramDriveFuel);
+		info.SetString("flagship scram drive fuel per hyperjump", to_string(flagshipScramDriveFuelDisplay));
+		double flagshipJumpDriveFuel = 200 + (((flagshipMass - 900) / 100) * 8);
+		int flagshipJumpDriveFuelDisplay = round(flagshipJumpDriveFuel);
+		info.SetString("flagship jump drive fuel per hyperjump", to_string(flagshipJumpDriveFuelDisplay));
```

Comment:
It took me a few times to spot the changes, but the optimization looks good. Thanks!

Was the entire block actually changed? Most of it Can't really spot any changes in.

Reply:
just very minor changes